cloudcix_primitives/utils.py

```diff
@@ -145,6 +145,34 @@ def load_pod_config(config_file=None, prefix=4000) -> Tuple[bool, Dict[str, Opti
 
     return True, config_data, f'{prefix + 10}: {messages[10]}'
 
+def write_rule(namespace: str, rule: Dict[str, Optional[Any]], user_chain: str) -> str:
+    """
+    Builds an ip/ip6 command string to write a rule to the provided chain.
+    :param namespace: network namespace to write the rule to.
+    :param rule: dictionary object containing rule configuration.
+    :param user_chain: nftables chain to add the rule to.
+    """
+    v = '' if str(rule['version']) == '4' else '6'
+
+    command = [f'ip{v} netns exec {namespace} nft add rule inet FILTER {user_chain} ip saddr {rule["source"]} ip daddr {rule["destination"]}']
+
+    if rule['protocol'] == 'icmp' and v == '4':
+        command.append('icmp type { echo-reply, destination-unreachable, echo-request, time-exceeded }')
+    elif rule['protocol'] == 'icmp' and v == '6':
```

Comment:
```suggestion
    elif rule['protocol'] == 'icmp' and str(rule['version']) == '6':
```

Changing so we are comparing with the same variable as in the IPv4 statement